bundles/pubsub/pubsub_admin_websocket/src/pubsub_websocket_topic_receiver.c

```diff
@@ -146,10 +147,13 @@ pubsub_websocket_topic_receiver_t* pubsub_websocketTopicReceiver_create(celix_bu
 
     receiver->uri = psa_websocket_createURI(scope, topic);
 
-    // Set receiver connection thread timeout.
-    // property is in ms, timeout value in us. (convert ms to us).
-    receiver->timeout = celix_bundleContext_getPropertyAsLong(ctx, PSA_WEBSOCKET_SUBSCRIBER_CONNECTION_TIMEOUT,
-                                                              PSA_WEBSOCKET_SUBSCRIBER_CONNECTION_DEFAULT_TIMEOUT) * 1000;
+    long timeoutMs = celix_bundleContext_getPropertyAsLong(ctx, PSA_WEBSOCKET_SUBSCRIBER_CONNECTION_TIMEOUT,
+                                                                PSA_WEBSOCKET_SUBSCRIBER_CONNECTION_DEFAULT_TIMEOUT);
+    receiver->timeoutUs = (useconds_t)((
+            ((timeoutMs < 0) || (timeoutMs > INT_MAX))            // Protect against under and overflow
```

Comment:
`INT_MAX * 1000` still overflows if useconds_t is something like uint32_t.

Reply:
Im sorry, I was trying to go to quick, thanks for spotting it!